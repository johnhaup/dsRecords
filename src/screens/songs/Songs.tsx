import firestore from '@react-native-firebase/firestore';
import React, { useEffect } from 'react';
import { StyleSheet, View } from 'react-native';
import Animated from 'react-native-reanimated';
import { SCREEN_WIDTH } from '../../constants/layout';

export const Songs = () => {
  const fetchSongs = async () => {
    try {
      const snapshot = await firestore().collection('songlist').get();
      const songs = snapshot.docs.map((doc) => doc.data());
      console.log(songs);
    } catch (error) {
      console.log(error);
    }
  };

  useEffect(() => {
    fetchSongs();
  }, []);

  return (
    <View style={styles.container}>
      <Animated.ScrollView
        testID={'@Songs/ScrollView'}
        horizontal
        pagingEnabled>
        <View style={styles.page} />
      </Animated.ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  page: {
    flex: 1,
    width: SCREEN_WIDTH,
  },
});
