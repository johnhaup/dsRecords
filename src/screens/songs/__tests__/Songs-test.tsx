import { cleanup, render } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import { act } from 'react-test-renderer';
import { Songs } from '../Songs';

afterEach(cleanup);

it('Matches snapshot', async () => {
  jest.useFakeTimers();
  const snap = render(<Songs />).toJSON();
  // Need these for state updates that occur in component
  await act(async () => {});
  await act(async () => {});
  expect(snap).toMatchSnapshot();
});
