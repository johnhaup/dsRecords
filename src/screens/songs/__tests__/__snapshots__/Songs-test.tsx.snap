// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "height": NaN,
          "width": 0,
        },
      ]
    }
  />
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListFooterComponent={
      <Spacer
        h={16}
      />
    }
    ListHeaderComponent={
      <Spacer
        h={16}
      />
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 8,
      }
    }
    testID="@Songs/FlatList"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "height": 16,
                "width": 0,
              },
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "height": 16,
                "width": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
